pkg/terraform/gather/ovirt/ip.go

```diff
@@ -2,13 +2,124 @@
 package ovirt
 
 import (
+	//"context"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/openshift/installer/pkg/asset/installconfig/ovirt"
 	"github.com/openshift/installer/pkg/terraform"
+	ovirtsdk4 "github.com/ovirt/go-ovirt"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	//"time"
 )
 
+const bootstrapSSHPort string = "22"
+
+func checkPortIsOpen(host string, port string) bool {
+	timeout := time.Second
+	conn, err := net.DialTimeout("tcp", net.JoinHostPort(host, port), timeout)
+	if err != nil {
+		logrus.Debugln(fmt.Sprintf("connection error: %v", err))
+	}
+	if conn != nil {
+		defer conn.Close()
+		return true
+	}
+	return false
```

Comment:
It is slightly confusing at first glance why the function is not short-circuited when there is an error. Also, I would rather separate the logic of setting up the defer from the logic of determining the return value.
```suggestion
	if err != nil {
		logrus.Debugln(fmt.Sprintf("connection error: %v", err))
		return false
	}
	if conn != nil {
		defer conn.Close()
	}
	return conn != nil
```

Reply:
done